bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/handler/RoomHandler.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.draytonwiser.internal.handler;
+
+import static org.openhab.binding.draytonwiser.internal.DraytonWiserBindingConstants.*;
+
+import javax.measure.quantity.Temperature;
+import javax.measure.quantity.Time;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.draytonwiser.internal.api.DraytonWiserApiException;
+import org.openhab.binding.draytonwiser.internal.model.DraytonWiserDTO;
+import org.openhab.binding.draytonwiser.internal.model.RoomDTO;
+import org.openhab.binding.draytonwiser.internal.model.RoomStatDTO;
+
+/**
+ * The {@link RoomHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andrew Schofield - Initial contribution
+ * @author Hilbrand Bouwkamp - Simplified handler to handle null data
+ */
+@NonNullByDefault
+public class RoomHandler extends DraytonWiserThingHandler<RoomDTO> {
+
+    private String name = "";
+
+    public RoomHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        name = (String) getConfig().get("name");
+    }
+
+    @Override
+    protected void handleCommand(final String channelId, final Command command) throws DraytonWiserApiException {
+        switch (channelId) {
+            case CHANNEL_CURRENT_SETPOINT:
+                if (command instanceof QuantityType) {
+                    setSetPoint((QuantityType<Temperature>) command);
```

Comment:
You could use the wildcard here and in `setSetPoint()` to eliminate the compiler warning.
```suggestion
                    setSetPoint((QuantityType<?>) command);
```

Reply:
Resolved in af8d624